scalableminds/webknossos
# Thumbnail improvements: persistent cache, configured intensity, float layers

- Replaced the TemporaryStore-based caching of thumbnails by a two-level caching (in-memory via AlfuCache and in postgres via new table)
 - Moved the parameter selection for thumbnails from datastore to wk side
 - Propagate intensityRange from dataset view configuration to thumbnail generation
 - Using that, and depending on dtype, perform intensity normalization
 - Also support float dataset thumbnails (here the intensityRange is a hard requirement, since float could otherwise go from negativeInfinity to positiveInfinity)

### TODO
 - [x] use intensityRange
 - [x] test cache clear
 - [x] test cleanup job
 - [x] store selected metadata in postgresql (mimetype, bbox, intensityrange?)
 - [ ] adapt postgres backups to skip this table?

### URL of deployed dev instance (used for testing):
- https://thumbnailcache.webknossos.xyz

### Steps to test:
- Visit the thumbnail route for a few datasets `http://localhost:9000/api/datasets/sample_organization/<datasetName>/layers/<layerName>/thumbnail`
- Only the first request should yield logging in backend like »sending request to thumbnail.jpg« (this generates the thumbnail on the datastore
- Subsequent should use cache
- Adapt intensity in view configuration of the dataset (in dataset settings page)
- Thumbnail should then look like that as well
- Also test this for a float layer.

### Issues:
- fixes #5089 
- fixes #6387 
- contributes to #6305 (though the front-end does not use this parameter yet)

------
